Package.swift

```diff
@@ -20,29 +19,31 @@ let package = Package(
             dependencies: [
                 "AsyncHTTPClient",
                 "AWSSignerV4",
-                "CAWSSDKOpenSSL",
                 "NIO",
                 "NIOHTTP1",
                 "NIOSSL",
                 "NIOTransportServices",
                 "NIOFoundationCompat",
                 "INIParser"
             ]),
-        .target(name: "AWSSignerV4", dependencies: ["CAWSSDKOpenSSL", "NIOHTTP1"]),
-        .target(name: "CAWSSDKOpenSSL", dependencies: []),
+        .target(name: "AWSSignerV4", dependencies: ["AWSCrypto", "NIOHTTP1"]),
+        .target(name: "AWSCrypto", dependencies: []),
+
+        .testTarget(name: "AWSCryptoTests", dependencies: ["AWSCrypto"]),
         .testTarget(name: "AWSSDKSwiftCoreTests", dependencies: ["AWSSDKSwiftCore", "NIOTestUtils"]),
         .testTarget(name: "AWSSignerTests", dependencies: ["AWSSignerV4"])
     ]
 )
 
-// switch for whether to use CAWSSDKOpenSSL to shim between OpenSSL versions
+// switch for whether to use swift crypto. Swift crypto requires macOS10.15 or iOS13.I'd rather not pass this requirement on
 #if os(Linux)
```

Comment:
#if canImport(CommonCrypto)
let useSwiftCrypto = false
#else 
let useSwiftCrypto = true
#endif

Reply:
But that isn't what we are testing for. We are testing for a platform. Maybe we should sync the platform tests instead

Reply:
It is a bit of a moot point in that they all mean the same thing

Reply:
I don't know: The point is implicit if we check for the operating system and is explicit if we check for the availability of the framework. For this reason I'm pro framework check.

Reply:
Yeah but CommonCrypto is the fallback. I'm not choosing CommonCrypto because it is available. I am choosing it because I don't want to enforce os version requirements on iOS and macOS. The os is the explicit choice, the framework is the implicit choice.

Reply:
fair point. Then let's work with if `os(Linux)` here. Let's maybe even make checks around macOS versions and replace everywhere else with 

```swift
#if !canImport(Crypto)
```

because this way we make sure it is really only used if `Crypto` is not available.

Reply:
I'm good with that

Reply:
done